### ISSUE TITLE
[Feature Request] WebTransport Support (HTTP3 over QUIC)

### ISSUE THREAD

**Issue by BxOxSxS** [erebe/wstunnel#191]
From [MDM web docs](https://developer.mozilla.org/en-US/docs/Web/API/WebTransport_API):
> The WebTransport API provides a modern update to WebSockets, transmitting data between client and server using HTTP/3 Transport. WebTransport provides support for multiple streams, unidirectional streams, and out-of-order delivery. It enables reliable transport via streams and unreliable transport via UDP-like datagrams.

In short: it would allow transporting data using QUIC with better performance than websocket over TCP

The downside of it is that some firewalls may block UDP on port 443, so there should be a switch (automatic?) to do fallback to websocket.

Regarding implementation, I have found these useful resources:
- https://github.com/hyperium/h3/tree/master/h3-webtransport
- https://github.com/BiagioFesta/wtransport
- https://github.com/kixelated/webtransport-rs

**Comment by bytejedi** [comment]
Some firewalls will also block quic because quic is also essentially udp.

**Comment by BxOxSxS** [comment]
I know there are some problems as these standards (QUIC, HTTP3 and WebTransport) are young, but they are being more and more adopted, so it's going to be only better. That's why I mentioned there should be a fallback. I think it's worth implementing, as there are certain scenarios where it's better. And I think over time there will be only more

**Comment by alou-S** [comment]
Most firewalls do block UDP. But due to HTTP/3 becoming more common there are a decent number of places that are allowing QUIC traffic through their often extremely restrictive firewalls, my university being one of them. Would be cool to actually see this feature implemented.

**Comment by liudongmiao** [comment]
RFC 9220: Bootstrapping WebSockets with HTTP/3

**Comment by ssrshz** [comment]
when will support http3 with better performance than websocket over TCP?
Looking forward to it

**Comment by ssrshz** [comment]
Dear @erebe ,
when will support http3 with better performance than websocket over TCP?Mainstream software already have supported http3,looking forward to it.

**Comment by deluxetiky** [comment]
Yes, we are looking forward to Quic support as well> It would be great to have QUIC support.

**Comment by zaczh** [comment]
Hi, thread.
If you are interested in proxy UDP via QUIC, you can check this repository: https://github.com/ferneast/masque-tunnel.
This is the standard way of proxying UDP via HTTP per the [RFC 9298](https://datatracker.ietf.org/doc/rfc9298/).
We’ve tested it with Apple’s Network Framework and confirmed that it’s compatible.
Disclaimer: The code is AI-generated and may have bugs. Feel free to submit issues if you encounter problems.

**Comment by erebe** [comment]
You can give a shot to the latest release v10.7.0. There is alpha/beta support for webtransport
